Composer/packages/server/src/services/project.ts

```diff
@@ -120,6 +120,11 @@ export class BotProjectService {
     return [...defaultProperties, ...userDefined];
   }
 
+  public static staticEntityResolver(projectId: string): string[] | undefined {
+    const contents = BotProjectService.getIndexedProjectById(projectId)?.luFiles.map((file) => file.content);
+    return flatten(contents);
```

Comment:
this would be the entire content of the lu files not the entities

Reply:
Yes. Since the extract luis entities is a async process. So I pass all contents of .lu file to LG server and do the parseFile operation. This way can simplify the code.

Reply:
I dont understand why dont you find all the entities here? why language server needs to do that? It seems like your code in language server calculates the entities every time there is a request for autocomplete, how is that more efficient?

Reply:
I think they are the same. Defining the function of calculating entites here. This resolver method is passed to LGServer. The actual computing stage is still occurred in the LGServer side. The bf-lu package for extract entities are optimized before. In LUServer, we also use the extracting entities for every request the server recieved.

Reply:
You need to cache the results, imagine you have lots of lu files, sending the contents of all those files and sending over to the language server, then extracting entities every time user type in a @ sign is not feasible imo.
@boydc2014 what do you think?

Reply:
yes, agree, we can't treat the extracted entities as "hot" data, instead, you should cache it somewhere and update it in a lower frequency than typing. Either with a explicit cache, or use something like debounce, throttle which might make your code simpler

Reply:
@cosmicshuai do you have what you need here to adjust this accordingly?

Reply:
@cwhitten Yes. I talked with Dong. I had a minor change in calculating entities. But there was some issue to push the commits yesterday.